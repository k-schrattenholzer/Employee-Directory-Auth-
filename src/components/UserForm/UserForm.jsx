import React, { useState } from 'react'
import { useForm } from '../../hooks/useForm.js'

export default function UserForm({ label, onSubmit }) {

  const { formState, formError, handleFormChange, setFormError } = useForm({
    email: '',
    password: ''
  });

  const [loading, setLoading] = useState(false);

  const handleSubmit = async (e) => {
    e.preventDefault();

    const { email, password } = formState;

    try {
      if( !email || password.length < 8)
        throw new Error("Can't complete your registration without an email, and a password with 8+ characters, pardner.");
        setLoading(true);
        await onSubmit(email, password);
    } catch(error) {
        setLoading(false);
        setFormError(error.message)
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <fieldset>
        <legend>{label}</legend>
        <section>
          <label htmlFor="email">email
          </label>
          <input 
            type="text"
            name="email"
            value={formState.email}
            onChange={handleFormChange}
            />
        </section>
        <section>
          <label htmlFor="password">password
          </label>
          <input 
            type="text"
            name="password"
            value={formState.password}
            onChange={handleFormChange}
            />
        </section>
        <button type='submit' disabled={loading}>
          {loading ? "checking in the back..." : label}
        </button>
        {formError && <p>{formError}</p> }
      </fieldset>
    </form>
  )
}
