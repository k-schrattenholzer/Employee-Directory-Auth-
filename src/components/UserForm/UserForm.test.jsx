import { render } from '@testing-library/react';
import UserForm from './UserForm.jsx'

it('should display a form for login stuff', () => {
  const { container } = render(<UserForm />);
  expect(container).toMatchSnapshot();
});