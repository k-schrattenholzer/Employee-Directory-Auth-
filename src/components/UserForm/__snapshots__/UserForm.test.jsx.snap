// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should display a form for login stuff 1`] = `
<div>
  <form>
    <fieldset>
      <legend />
      <section>
        <label
          for="email"
        >
          email
        </label>
        <input
          name="email"
          type="text"
          value=""
        />
      </section>
      <section>
        <label
          for="password"
        >
          password
        </label>
        <input
          name="password"
          type="text"
          value=""
        />
      </section>
      <button
        type="submit"
      />
      
    </fieldset>
  </form>
</div>
`;
